### PR TITLE
Allow to map directly from JSON string

### LINKED ISSUE #43: Allow to map directly from JSON string
As it is quite common to have a JSON in string format. It would be helpful for users of the library to map from the string instead of an object. This would reduce their boiler plate code when using JsonMapper together with an HTTP JSON API.

The desired solution should add some convenience methods such as `mapObjectFromString` and `mapArrayFromString` to the `JsonMapperInterface` and it's implementation.

### PR DESCRIPTION
**Pull request by gonzalezlopezjm**
Fixes #43

### CODE REVIEW

**Review by DannyvdSluijs — COMMENTED** [review]
Thank for you PR! I had some remarks on which Im curious. If you want you can address them if not I can pick up and continue the PR later this week.

File: src/JsonMapper.php [hunk 1]
```diff
@@ -98,6 +98,28 @@ public function mapArray(array $json, object $object): array
...
+        $data = \json_decode($json);
+
+        if (!$data) {
+	        $data = new \stdClass();
+        }
+
+    	$this->mapObject($data, $object);
+    }
+
+    public function mapArrayFromString(array $json, object $object): array
```
[the remark this answers is not in the document]
Comment (gonzalezlopezjm) [reply src/JsonMapper.php:112]: I understand that was an array of JSON strings and not a full JSON, I will change it, no problem

File: tests/Integration/JsonMapperTest.php [hunk 2]
```diff
@@ -212,6 +240,22 @@ public function testItCanMapAnArrayOfObjects(): void
         self::assertSame('two', $result[1]->getName());
     }
 
+	public function testItCanMapAnArrayOfString(): void
+	{
+		// Arrange
+		$mapper = (new JsonMapperFactory())->bestFit();
+		$object = new SimpleObject();
+		$jsonString = ['{"name": "one"}', '{"name": "two"}'];
```
[the remark this answers is not in the document]
Comment (gonzalezlopezjm) [reply tests/Integration/JsonMapperTest.php:248]: As the previous comment, I will change it

File: src/JsonMapper.php [hunk 3]
```diff
@@ -98,6 +98,28 @@ public function mapArray(array $json, object $object): array
         return $results;
     }
 
+    public function mapObjectFromString(string $json, object $object): void
+    {
+        $data = \json_decode($json);
```
[the remark this answers is not in the document]
Comment (gonzalezlopezjm) [reply src/JsonMapper.php:103]: I thought that launch an exception was a worse solution that use an empty data source (the way that I chose), but I haven't any problem to go in this way

**Comment by DannyvdSluijs** [comment]
Closed in favour of #46.

**Comment by gonzalezlopezjm** [comment]
I'm happy, it's my first successful open source contribution

**Pull request closed without merging**